sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs

```diff
@@ -87,17 +87,24 @@ public virtual async ValueTask DisposeAsync()
         }
 
         /// <summary>
-        /// Can be used for mocking.
+        /// Gets the metrics associated with this <see cref="ServiceBusClient"/> instance. In order to use this property,
+        /// <see cref="ServiceBusClientOptions.EnableTransportMetrics"/> must be set to <value>true</value>.
         /// </summary>
-        protected ServiceBusClient()
-        {
-        }
+        public ServiceBusTransportMetrics TransportMetrics
+            => Connection.InnerClient.Metrics ?? throw new InvalidOperationException("Transport metrics are not enabled. To enable transport metrics, set the EnableTransportMetrics property on the ServiceBusClientOptions.");
```

Comment:
Should we consider cloning the instance to ensure that data doesn't mutate for callers?   As a caller, I would expect that anything you return to me will be stable when I'm reading it.

Reply:
We could, but then I think we'd need to switch to a method rather than property, which might be okay.